elements/barcode-reader/src/barcode-reader.js

```diff
@@ -340,30 +360,35 @@ class BarcodeReader extends LitElement {
   }
 
   async start() {
-    this.shadowRoot.getElementById("render").addEventListener("click", () => {
+    this.shadowRoot.querySelector(".render").addEventListener("click", () => {
       if (
-        this.shadowRoot.getElementById("render").innerHTML ===
+        this.shadowRoot.querySelector(".render").innerHTML ===
         "Initialize scanner"
       ) {
         this._control();
       }
     });
   }
   async _renderVideo() {
-    let video = this.shadowRoot.getElementById("hidden");
-    let button = this.shadowRoot.getElementById("render");
-    let extraButtons = this.shadowRoot.getElementById("hidden2");
+    let video = this.shadowRoot.querySelector(".hidden");
+    let button = this.shadowRoot.querySelector(".render");
+    let extraButtons = this.shadowRoot.querySelector(".hidden2");
     video.style.display = "none";
-    this.shadowRoot.getElementById("render").addEventListener("click", () => {
+    this.shadowRoot.querySelector(".render").addEventListener("click", () => {
```

Comment:
`this.showScanner = !this.showScanner;` then in constructor showScanner is false. In the properties block it's a boolean / attribute binding to match dash case. Then in an `updated` life-cycle you'd see if showScanner is true, then run your logic for truthy. If it's false, run falsey logic.

The reason to do this is to enhance the statefulness of your element. Right now without user interaction (click) there is no way to active this. Perhaps it's used in a larger application or part of another element. It would be useful for THAT element to dictate when the camera is shown. Without an attribute to change we're not able to propagate data flow down into the element to modify it and instead have to fake it via other events.